### PR TITLE
fix: throw in `onSend` causes fastify to not respond

### PR DESCRIPTION
**Pull request by AyoubElk**
fixes the issue described in #2692 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/hooks.js [hunk 1]
```diff
@@ -199,7 +199,12 @@ function onSendHookRunner (functions, request, reply, payload, cb) {
       return
     }
 
-    const result = functions[i++](request, reply, payload, next)
+    let result
+    try {
+      result = functions[i++](request, reply, payload, next)
+    } catch (error) {
+      next(error)
```
Comment (mcollina) [inline lib/hooks.js:206]:
```suggestion
      next(error)
      return
```

**Review by mcollina — COMMENTED** [review]
thanks, just one nit!

**Review by RafaelGSS — APPROVED** [review]
LGTM.

**Comment by AyoubElk** [comment]
This issue mentions the `onSend` hook, I'm still not very familiar with the codebase but I think maybe a similar fix is needed in `hookRunner` and `hookIterator` functions which are used for other hooks?

Let me know if you agree, and if I missed some other function that needs the fix as well.

**Review by delvedor — CHANGES_REQUESTED** [review]
This should be fixed also for every other hook :)
https://github.com/fastify/fastify/blob/2f46d9a4b957a9b1c30d380eb2df9e2fd374a4ce/lib/hooks.js#L162

**Review by Eomm — APPROVED** [review]

**Comment by AyoubElk** [comment]
Just found out that preParsing hooks have a separate runner, I added a fix for that as well.

**Comment by mcollina** [comment]
@delvedor ok for you now?

**Review by Ethan-Arrowood — APPROVED** [review]

**Review by fox1t — APPROVED** [review]

**Review by delvedor — APPROVED** [review]
LGTM

**Pull request merged**